payments/utils.py

```diff
@@ -225,9 +225,32 @@ def generate_order_number() -> str:
     return b.decode("utf8")
 
 
+def resolve_area(order: Order):
+    lease_order = (
+        order
+        if order.order_type == OrderType.LEASE_ORDER
+        else (
+            order.lease.orders.filter(
+                status__in=OrderStatus.get_paid_statuses(),
```

Comment:
If I understand correctly, it should be quite rare (or practically impossible) for this query to return more than one order, but just in case, and to make the result deterministic, maybe sort the list and use the most recent paid LEASE_ORDER?

Reply:
Good point. As you said it _shouldn't_ be the case, but I'll add it just in case.